Kernel/Arch/aarch64/MMU.cpp

```diff
@@ -244,4 +244,25 @@ void init_page_tables()
     activate_mmu();
 }
 
+void unmap_identity_map()
+{
+    auto start_of_physical_memory = FlatPtr(START_OF_NORMAL_MEMORY);
+
+    u64 level0_idx = (start_of_physical_memory >> 39) & 0x1FF;
+    u64 level1_idx = (start_of_physical_memory >> 30) & 0x1FF;
+
+    u64* level1_table = (u64*)page_tables_phys_start;
+
+    auto level2_table = FlatPtr(descriptor_to_pointer(level1_table[level0_idx]));
+    if (!level2_table)
+        panic_without_mmu("Could not find table!"sv);
+
+    // NOTE: descriptor_to_pointer returns a physical address, but we want to unmap that range
+    //       so, it the pointer must be converted to a virtual address by adding KERNEL_MAPPING_BASE.
+    level2_table += KERNEL_MAPPING_BASE;
+
+    // Unmap the complete identity map
+    ((u64*)level2_table)[level1_idx] = 0;
```

Comment:
I am not familiar with ARM64 paging, so feel free to ignore this if I'm wrong.

Here you are setting to zero a single entry in a lvl2 table. I assume this is to quickly unmap 1GB, which is also how much RAM the raspi3 has.

Can we add an assert on the physical RAM size, or at least a FIXME? Just in case someone decides to port serenity also to other platforms with more or less RAM

Reply:
Yes, you are correct. This is indeed what is happening, quickly unmapping the identity maps. However, we only map the kernel image, and the MMIO range, not the complete 1GB of RAM. So I'm not sure if we need to assert something here.